lib/news_scraper/extractors/google_news_rss.rb

```diff
@@ -39,18 +37,18 @@ def google_urls_from_resp(body)
         end
       end
 
-      def extract_article_uris(google_urls)
+      def extract_article_urls(google_urls)
         google_urls.map do |google_url|
-          regex = google_url.match(/&url=https?:\/\/(.*)/)
-          regex.nil? ? nil : regex[1]
+          regex = google_url.match(%r(&url=(?<url>https?://.*)))
+          regex.nil? ? nil : regex['url']
         end.compact.uniq
       end
 
-      def write_to_temp(article_uris)
-        filename = "#{query.downcase.gsub(/\s/, '_')}_#{Time.now.to_i}.txt"
-        puts "Writing article uris to #{filename}"
-        File.open(File.join(TMP_DIR, filename), 'w') do |file|
-          article_uris.each { |uri| file.puts uri }
+      def write_to_temp(article_urls)
+        filename = "#{@query.downcase.gsub(/\s/, '_')}_#{Time.now.to_i}.txt"
+        puts "Writing article urls to #{filename}"
+        File.open(File.join(ARTICLE_URLS_DIR, filename), 'w') do |file|
```

Comment:
Well right now it's just the projects tmp folder, which is gitignored.

On Sep 1, 2016 9:50 PM, "Julian Nadeau" <notifications@github.com> wrote:

> In lib/news_scraper/extractors/google_news_rss.rb
> <https://github.com/richardwu/news_scraper/pull/6#discussion_r77283077>:
>
> >          end.compact.uniq
> >        end
> >
> > -      def write_to_temp(article_uris)
> > -        filename = "#{query.downcase.gsub(/\s/, '_')}_#{Time.now.to_i}.txt"
> > -        puts "Writing article uris to #{filename}"
> > -        File.open(File.join(TMP_DIR, filename), 'w') do |file|
> > -          article_uris.each { |uri| file.puts uri }
> > +      def write_to_temp(article_urls)
> > +        filename = "#{@query.downcase.gsub(/\s/, '_')}_#{Time.now.to_i}.txt"
> > +        puts "Writing article urls to #{filename}"
> > +        File.open(File.join(ARTICLE_URLS_DIR, filename), 'w') do |file|
>
> cool, how long is the "tmp" folder? Is it really a cache?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/richardwu/news_scraper/pull/6/files/1fff18f3aa533a4151a86d9d3bbaa3f0cb80189d#r77283077>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKEu8tBXYBnNXeSq2LmqdSVvezSpww38ks5ql4D5gaJpZM4JzP8F>
> .
>